docs/forms/FORM_ERRORS.md

```diff
@@ -100,38 +78,45 @@ export interface ErrorMessageObject {
-The error message structure will be defined in the implementation of __validate__ method.
-As __validate__ function should return a [ValidationErrors](https://angular.io/api/forms/ValidationErrors) object, which is a map of custom objects (with type _any_), we can prepare the returned object for the store of error messages. This will ease the process of adding the errors in the store.
-We have to make sure that we are providing the __htmlElementId__ for the errors in the store which is matching the __html field__.
-For this, the presenter is receiving an __id__ as input and for each field we are concatenating the __id__ with the __formControlName__. As the container is setting a __unique id__ we are sure that we have uniques html ids for the form fields.
-The object returned by the __validate__ is the error object which is propagated to the container.
+The presenter has to implement the [Validator](https://angular.io/api/forms/NG_VALIDATORS) or the [AsyncValidator](https://angular.io/api/forms/NG_ASYNC_VALIDATORS) interface
+in order to give us the possibility of defining the error object that will be returned by the form.
 
-There are 2 types of validators (see [Form Validation](./FORM_VALIDATION.md)), 2 categories of error messages:
+The error message structure will be defined in the implementation of the `validate` function.
 
-- one for __custom errors__ - set on the container
-- one for __primitive errors__ - computed in the presenter.
+As the `validate` function should return a [ValidationErrors](https://angular.io/api/forms/ValidationErrors) object, which is a map of custom objects (of type `any`),
+we can prepare the returned object for the store of error messages. This will ease the process of adding the errors in the store.
 
-<a name="customerrors"></a>
+We have to make sure that we provide the `htmlElementId` of the errors in the store that match the __HTML fields__.
+For this, the presenter receives an `id` as input and for each field we are concatenating, the `id` with the `formControlName`.
+Since the container sets a __unique id__, we are sure to have unique HTML identifiers for the form fields.
 
-#### Custom errors
+The object returned by the `validate` function is the error object that is propagated to the container.
+
+#### Categories of error messages
 
-They are returned by __custom validators__ and have the type [CustomErrors](https://github.com/AmadeusITGroup/otter/blob/main/packages/@o3r/forms/src/core/custom-validation.ts) defined in __@o3r/forms__.
-This one is using _customErrors_ key with an array of __ErrorMessageObject__ which has to contain all the custom errors for a form control or group.
+There are two types of validators (see [Form Validation](./FORM_VALIDATION.md)) and therefore two categories of error messages:
+
+- __Custom error__ - set in the container
```

Comment:
I think it could be worth to add something about why setting an error in the presenter or in the container.

Reply:
In the form validation file, the difference between the two validators is explained, should I re explain the difference here ?